solver/src/naive_solver/multi_order_solver.rs

```diff
@@ -237,8 +236,11 @@ fn compute_uniswap_out(shortage: &TokenContext, excess: &TokenContext) -> U256 {
 /// of tokens to be sent to the pool.
 /// Taken from: https://github.com/Uniswap/uniswap-v2-periphery/blob/4123f93278b60bcf617130629c69d4016f9e7584/contracts/libraries/UniswapV2Library.sol#L53
 ///
-fn compute_uniswap_in(out: U256, shortage: &TokenContext, excess: &TokenContext) -> U256 {
-    U256::from(1000) * out * excess.reserve / (U256::from(997) * (shortage.reserve - out)) + 1
+fn compute_uniswap_in(out: U256, shortage: &TokenContext, excess: &TokenContext) -> Option<U256> {
+    let numerator = 1000 * u256_to_big_int(&out) * u256_to_big_int(&excess.reserve);
+    let denominator = 997 * u256_to_big_int(&(shortage.reserve - out));
+    let result: BigInt = numerator / denominator + 1;
+    bigint_to_u256(&result).ok()
```

Comment:
Why not just use the `checked_${operator}` methods directly here?

Reply:
My thinking was that using bigint would allow temporary overflow that is resolved by the division. Not sure which way is better.

Reply:
So the Uniswap contract uses `uint112` with checked operations, so overflowing the `BigInt` would probably result in a revert on-chain.